crates/client/gateway/src/client/methods.rs

```diff
@@ -0,0 +1,90 @@
+use mp_block::{BlockId, BlockTag};
+use serde::{Deserialize, Serialize};
+
+use crate::error::SequencerError;
+
+use super::{builder::FeederClient, request_builder::RequestBuilder};
+
+use mp_gateway::{
+    block::{BlockProvider, PendingBlockProvider, ProviderMaybePendingBlock},
+    state_update::{PendingStateUpdateProvider, ProviderMaybePendingStateUpdate, StateUpdateProvider},
+};
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct StateUpdateWithBlock {
+    pub state_update: StateUpdateProvider,
+    pub block: BlockProvider,
+}
+
+impl FeederClient {
+    pub async fn get_block(&self, block_id: BlockId) -> Result<ProviderMaybePendingBlock, SequencerError> {
+        let request = RequestBuilder::new(&self.client, self.feeder_gateway_url.clone())
+            .add_uri_segment("get_block")
+            .unwrap()
+            .with_block_id(block_id);
+
+        match block_id {
+            BlockId::Tag(BlockTag::Pending) => {
+                Ok(ProviderMaybePendingBlock::Pending(request.send_get::<PendingBlockProvider>().await?))
+            }
+            _ => Ok(ProviderMaybePendingBlock::Block(request.send_get::<BlockProvider>().await?)),
+        }
+    }
+
+    pub async fn get_state_update(&self, block_id: BlockId) -> Result<ProviderMaybePendingStateUpdate, SequencerError> {
+        let request = RequestBuilder::new(&self.client, self.feeder_gateway_url.clone())
+            .add_uri_segment("get_state_update")
+            .unwrap()
+            .with_block_id(block_id);
+
+        match block_id {
+            BlockId::Tag(BlockTag::Pending) => {
+                Ok(ProviderMaybePendingStateUpdate::Pending(request.send_get::<PendingStateUpdateProvider>().await?))
+            }
+            _ => Ok(ProviderMaybePendingStateUpdate::Update(request.send_get::<StateUpdateProvider>().await?)),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use mp_block::BlockTag;
+    use starknet_core::types::Felt;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_get_block() {
```

Comment:
Why no `assert`? It seems you are testing for block 0, maybe we could check the contents of the block too, or at the very least asserts its block number?